docs/spec_draft.md

```diff
@@ -941,6 +942,62 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dynamic_slice
+
+### Semantics
+
+Extracts a slice of shape `slice_sizes` from the `operand` tensor starting at
+`start_indices` and produces a `result` tensor.
+
+More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` such that:
+  * `jd = effective_start_indices[d] + id`.
+  * `effective_start_indices[d] = clamp(start_indices[d], 0, dim(operand, d) - `
+    `slice_sizes[d])`.
+
+### Inputs
+
+| Name            | Type                                                    |
+|-----------------|---------------------------------------------------------|
+| `operand`       | tensor of any supported type                            |
+| `start_indices` | variadic number of 0-dimensional tensors of type `si64` |
+| `slice_sizes`   | 1-dimensional tensor constant of type `si64`            |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) size(`start_indices`) $=$ size(`slice_sizes`) $=$ rank(`operand`).
+  * (C3) `slice_sizes[k]` $\in$ [0, dim(`operand`, `k`)) for all `k` $\in$
+    [0, rank(`operand`)).
+  * (C4) shape(`result`) $=$ `slice_sizes`.
+
+### Examples
+
+```mlir
+// %operand: [
+//            [0, 0, 0, 0],
+//            [0, 0, 1, 1],
+//            [0, 0, 1, 1],
+//            [0, 0, 0, 0]
+//           ]
+// %start_indices0: 1
+// %start_indices1: 2
```

Comment:
How about changing `%start_indices0` to `-1` and `%start_indices1` to `3` to illustrate how effective_start_indices work?